datasync/chaindatafetcher/chaindata_fetcher.go

```diff
@@ -354,29 +353,75 @@ func (f *ChainDataFetcher) updateCheckpoint(num int64) error {
 	return nil
 }
 
-func (f *ChainDataFetcher) updateGauge(insert func(chainEvent blockchain.ChainEvent), gauge metrics.Gauge) func(chainEvent blockchain.ChainEvent) {
-	return func(chainEvent blockchain.ChainEvent) {
+func getInsertionTimeGauge(reqType cfTypes.RequestType) (metrics.Gauge, error) {
+	switch reqType {
+	case cfTypes.RequestTypeTransaction:
+		return txsInsertionTimeGauge, nil
+	case cfTypes.RequestTypeTokenTransfer:
+		return tokenTransfersInsertionTimeGauge, nil
+	case cfTypes.RequestTypeContract:
+		return contractsInsertionTimeGauge, nil
+	case cfTypes.RequestTypeTrace:
+		return tracesInsertionTimeGauge, nil
+	default:
+		return nil, fmt.Errorf("the request type is not supported. type: %v", reqType)
+	}
+}
+
+func (f *ChainDataFetcher) updateInsertionTimeGauge(insert HandleChainEventFn) HandleChainEventFn {
+	return func(chainEvent blockchain.ChainEvent, reqType cfTypes.RequestType) error {
 		now := time.Now()
-		insert(chainEvent)
+		if err := insert(chainEvent, reqType); err != nil {
+			return err
+		}
 		elapsed := time.Since(now)
-		gauge.Update(elapsed.Milliseconds())
+		if gauge, err := getInsertionTimeGauge(reqType); err != nil {
+			// ignore handling error since the metrics does not affect the chaindatafetcher logic.
+			// instead, the following log is left.
+			logger.Warn("getInsertionTimeGauge is failed", "err", err, "type", reqType)
+		} else {
+			gauge.Update(elapsed.Milliseconds())
+		}
+		return nil
 	}
 }
 
-func (f *ChainDataFetcher) retryFunc(insert func(blockchain.ChainEvent) error, gauge metrics.Gauge) func(blockchain.ChainEvent) {
-	return func(event blockchain.ChainEvent) {
+func getInsertionRetryGauge(reqType cfTypes.RequestType) (metrics.Gauge, error) {
```

Comment:
Same comment for error handling logic.

Reply:
Agreed. Reflected on a9b659884694577dba20dad99f4c46673df1aab3.